codebase-pipelines/buildspec-deploy.yml

```diff
@@ -3,16 +3,67 @@ version: 0.2
 env:
   variables:
     DEPLOY_TIMEOUT: 1800
+  parameter-store:
+    SLACK_TOKEN: /codebuild/slack_oauth_token
 
 phases:
   install:
     commands:
+      - echo "Starting deployment script for ${SERVICE} service"
       - pip install yq --quiet
+      - echo "Installing platform-helper"
+      - pip install dbt-platform-helper --quiet
+      - platform-helper --version
+      - echo "Installing regclient"
+      - curl -s -L https://github.com/regclient/regclient/releases/latest/download/regctl-linux-amd64 > /usr/local/bin/regctl
+      - chmod +x /usr/local/bin/regctl
 
   build:
     commands:
       - set -e
 
+      # Extract timestamp from image config and check if it exists
+      - aws ecr get-login-password --region ${AWS_REGION} | docker login --username AWS --password-stdin ${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_REGION}.amazonaws.com
+
+      # construct Slack message env vars
+      - |
+        SLACK_REF=$(regctl image config "${REPOSITORY_URL}:${IMAGE_TAG}" | jq -r '.config.Labels."uk.gov.trade.digital.build.timestamp"')
+        if [ "$SLACK_REF" = "null" ] || [ -z "$SLACK_REF" ]; then
+          echo "Image contains no timestamp label"
+          exit 1
+        fi
+        echo "Found image timestamp $SLACK_REF"
+
+      - UPPERCASE_SERVICE=$(echo "${SERVICE}" | tr '[:lower:]' '[:upper:]')
+      - UPPERCASE_TAG=$(echo "${IMAGE_TAG}" | tr '[:lower:]' '[:upper:]')
+
+        # Extract the pipeline name from CODEBUILD_INITIATOR default env var
+      - |
+        PIPELINE_NAME=""
+
+        if [[ "${CODEBUILD_INITIATOR}" == codepipeline/* ]]; then 
+          PIPELINE_NAME=$(echo "${CODEBUILD_INITIATOR}" | cut -d'/' -f2) 
+          echo "Pipeline name is ${PIPELINE_NAME}"
+        else 
+          echo "ERROR: Build not triggered by CodePipeline." 
+          exit 1 
+        fi
+
+      # Construct the pipeline execution URL
+      - |
+        PIPELINE_EXECUTION_URL="https://${AWS_REGION}.console.aws.amazon.com/codesuite/codepipeline/pipelines/${PIPELINE_NAME}/executions/${PIPELINE_EXECUTION_ID}" 
+
+        echo "Pipeline execution URL: ${PIPELINE_EXECUTION_URL}"
+
+      - |
+        BUILD_ID_PREFIX=$(echo $CODEBUILD_BUILD_ID | cut -d':' -f1)
+        echo "BUILD_ID_PREFIX - ${BUILD_ID_PREFIX}"
+
+      - |
```

Comment:
This can be just a single line item. `- MESSAGE=...`

Reply:
FYI - you should be able to just surround the line in single quotes. It's because it thinks it's a map rather than a string. Surrounding in single quotes will make the YAML parser treat it as a string. 

But happy for you to leave it as a multi-line quote in this case as it's probably neater.